src/mm/checkpoints.c

```diff
@@ -219,6 +252,90 @@ void *log_state(int lid) {
 
 
 
+bool scan_recoverable_area(malloc_area **m_area, void **ptr, int *bitmap_blocks, int *restored_areas, malloc_state **recoverable_state) {
```

Comment:
why pointer of pointer?
```suggestion
bool check_marea_rebuild(malloc_area *m_area, void **ptr, int *bitmap_blocks, int *restored_areas, malloc_state **recoverable_state) {
```